app/controllers/requested_accounts_controller.rb

```diff
@@ -62,15 +74,31 @@ def create_accounts
     end
   end
 
+  def create_all_accounts
+    raise_unauthorized_exception unless user_signed_in? && current_user.admin?
+    @courses = all_requested_accounts
+    @results = RequestedAccount.all.inject([]) do |results, account|
+      results << create_account(account)
+    end
+
+    render :index
+  end
+
   private
 
+  def all_requested_accounts
+    RequestedAccount.all.group_by { |account| account.course.title }
```

Comment:
Not likely to ever be a performance problem, this gets rid of the N+1 query.

```suggestion
    RequestedAccount.all.includes(:course).group_by { |account| account.course.title }
```